### PR TITLE
feat: update account email when TIS changes

### PR DESCRIPTION
**Pull request by Judge40**
Update the email address associated with a user account when the TIS contact details are changed.
The email should only be updated if it is not in use by any other user.

Exceptions should be throw in two cases
 - Where multiple accounts exist for a given PersonId
 - Where the new email address is already in use

Cache the users retrieved from cognito as the lookup is costly.

TIS21-5907
TIS21-5940

### CODE REVIEW

File: src/test/java/uk/nhs/tis/trainee/usermanagement/service/UserAccountServiceIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,132 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright 2023 Crown Copyright (Health Education England)
```
Comment (ReubenRobertsHEE) [inline src/test/java/uk/nhs/tis/trainee/usermanagement/service/UserAccountServiceIntegrationTest.java:4]: You're living in the past mate 😁

Comment (Judge40) [reply src/test/java/uk/nhs/tis/trainee/usermanagement/service/UserAccountServiceIntegrationTest.java:4]: I'm not well up on my copyright laws, given this is lifted and unchanged from another service as written in 2023... do we update the copyright here or leave it then same 🤔

**Comment by ReubenRobertsHEE** [comment]
It all looks good to me, but one thing I was wondering: when we update the users login details, surely we should send them a notification that it has happened? Maybe its an interation+1 thing, but I'm thinking posting onto a user-login-details-updated queue and setting the notification service to listen to that and generate an appropriate mail.

**Review by ReubenRobertsHEE — APPROVED** [review]

**Comment by Judge40** [comment]
> It all looks good to me, but one thing I was wondering: when we update the users login details, surely we should send them a notification that it has happened? Maybe its an interation+1 thing, but I'm thinking posting onto a user-login-details-updated queue and setting the notification service to listen to that and generate an appropriate mail.

Yeah, MVP + 1 🙈 
I'm absolutely keen we do that ASAP and I'll be creating a follow-up ticket. As you say it's not complex to send a simple email to both the old and new email addresses - there is also a need to notify/inform on the Admins UI side that could be rolled in to the same ticket.

**Pull request merged**